packages/dev/core/src/Rendering/fluidRenderer/fluidRenderer.ts

```diff
@@ -0,0 +1,426 @@
+import type { Scene } from "core/scene";
+import type { Engine } from "core/Engines/engine";
+import type { FloatArray, Nullable } from "core/types";
+import type { Observer } from "core/Misc/observable";
+import type { Camera } from "core/Cameras/camera";
+import type { IParticleSystem } from "core/Particles/IParticleSystem";
+
+import type { FluidRenderingObject } from "./fluidRenderingObject";
+import { FluidRenderingObjectParticleSystem } from "./fluidRenderingObjectParticleSystem";
+import { FluidRenderingTargetRenderer } from "./fluidRenderingTargetRenderer";
+import { FluidRenderingObjectCustomParticles } from "./fluidRenderingObjectCustomParticles";
+import { FluidRenderingDepthTextureCopy } from "./fluidRenderingDepthTextureCopy";
+
+import "../../Shaders/fluidRenderingParticleDepth.vertex";
+import "../../Shaders/fluidRenderingParticleDepth.fragment";
+import "../../Shaders/fluidRenderingParticleThickness.vertex";
+import "../../Shaders/fluidRenderingParticleThickness.fragment";
+import "../../Shaders/fluidRenderingParticleDiffuse.vertex";
+import "../../Shaders/fluidRenderingParticleDiffuse.fragment";
+import "../../Shaders/fluidRenderingBilateralBlur.fragment";
+import "../../Shaders/fluidRenderingStandardBlur.fragment";
+import "../../Shaders/fluidRenderingRender.fragment";
+
+/**
+ * An object rendered as a fluid.
+ * It consists of the object itself as well as the render target renderer (which is used to generate the textures (render target) needed for fluid rendering)
+ */
+export interface IFluidRenderingRenderObject {
+    /** object rendered as a fluid */
+    object: FluidRenderingObject;
+    /** target renderer used to render the fluid object */
+    targetRenderer: FluidRenderingTargetRenderer;
+}
+
+/**
+ * Class responsible for fluid rendering.
+ * It is implementing the method described in https://developer.download.nvidia.com/presentations/2010/gdc/Direct3D_Effects.pdf
+ */
+export class FluidRenderer {
+    /** @hidden */
+    public static _SceneComponentInitialization: (scene: Scene) => void = (/*_*/) => {
+        throw `FluidRendererSceneComponent needs to be imported before as it contains a side-effect required by your code.`;

```

Comment:
The easiest way would be to simply paste the `FluidRenderSceneComponent` code into `FluidRenderer`...

Since `FluidRenderer` can't work without it, this makes sense, and it's less of a burden on the user who won't have to import the `FluidRendererSceneComponent` for its side effects.

> it might need to externalize the instantiation to prevent the cyclic ref :-(

We would have problems with both:
```typescript
Scene.prototype.enableFluidRenderer = function (): Nullable<FluidRenderer> {
    if (this._fluidRenderer) {
        return this._fluidRenderer;
    }

    this._fluidRenderer = new FluidRenderer(this);

    return this._fluidRenderer;
};
```
and:
```typescript
FluidRenderer._SceneComponentInitialization = (scene: Scene) => {
    let component = scene._getComponent(SceneComponentConstants.NAME_FLUIDRENDERER) as FluidRendererSceneComponent;
    if (!component) {
        component = new FluidRendererSceneComponent(scene);
        scene._addComponent(component);
    }
};
```

Reply:
Grouping them makes sense to me as it would prevent usage mistake ? what do you think ?

Reply:
Yes makes sense => I have updated the sources.